tt_metal/fabric/hw/inc/edm_fabric/1d_fabric_constants.hpp

```diff
@@ -170,12 +170,23 @@ constexpr size_t sender_3_completed_packet_header_cb_size_headers =
 constexpr size_t sender_4_completed_packet_header_cb_address = get_compile_time_arg_val(MAIN_CT_ARGS_START_IDX + 54);
 constexpr size_t sender_4_completed_packet_header_cb_size_headers =
     get_compile_time_arg_val(MAIN_CT_ARGS_START_IDX + 55);
-constexpr size_t is_2d_fabric = get_compile_time_arg_val(MAIN_CT_ARGS_START_IDX + 56);
-constexpr size_t my_direction = get_compile_time_arg_val(MAIN_CT_ARGS_START_IDX + 57);
-
-constexpr size_t num_eth_ports = get_compile_time_arg_val(MAIN_CT_ARGS_START_IDX + 58);
-
-constexpr size_t SPECIAL_MARKER_0_IDX = MAIN_CT_ARGS_START_IDX + 59;
+constexpr std::array<bool, MAX_NUM_SENDER_CHANNELS> is_sender_channel_serviced =
+    fill_array_with_next_n_args<bool, MAIN_CT_ARGS_START_IDX + 56, MAX_NUM_SENDER_CHANNELS>();
```

Comment:
because `MAX_NUM_SENDER_CHANNELS` is configurable, then I think it is preferred to do something like
```suggestion
constexpr size_t sender_channel_serviced_args_idx = MAX_NUM_SENDER_CHANNELS = 56;
constexpr std::array<bool, MAX_NUM_SENDER_CHANNELS> is_sender_channel_serviced =
    fill_array_with_next_n_args<bool, MAIN_CT_ARGS_START_IDX + 56, MAX_NUM_SENDER_CHANNELS>();

constexpr size_t receiver_channel_serviced_args_idx = sender_channel_serviced_args_idx + is_sender_channel_serviced.size();
constexpr std::array<bool, MAX_NUM_RECEIVER_CHANNELS> is_receiver_channel_serviced =
    fill_array_with_next_n_args<bool, receiver_channel_serviced_args_idx, MAX_NUM_RECEIVER_CHANNELS>();

constexpr size_t NEXT_ARGS_BLOCK_IDX = receiver_channel_serviced_args_idx + is_receiver_channel_serviced.size();
```
(or if there's a prettier way, then go with that instead)

Then the following args are indexing with:
```
... = get_compile_time_arg_val(NEXT_ARGS_BLOCK_IDX);
... = get_compile_time_arg_val(NEXT_ARGS_BLOCK_IDX + 1);
... = get_compile_time_arg_val(NEXT_ARGS_BLOCK_IDX + 2);
...
```